plugins/restful/RestfulEntityBase.php

```diff
@@ -655,7 +660,19 @@ protected function setPropertyValues(EntityMetadataWrapper $wrapper, $null_missi
 
       $field_value = $this->propertyValuesPreprocess($property_name, $request[$public_field_name], $public_field_name);
 
-      $wrapper->{$property_name}->set($field_value);
+      // Support sub property saving.
+      if ($sub_property_name) {
+        if (!$wrapper->{$property_name}->value()) {
+          // Init field.
+         $wrapper->{$property_name}->set(array($sub_property_name => ''));
```

Comment:
Is it safe to init this to a string? Should this be a NULL?

Do we need to init the field?